### PR TITLE
Refactor sortByLoc to use @glimmer/syntax' sortByLoc function

### PR DESCRIPTION
**Pull request by dcyriller**
## Description

Starting from v0.76.0, a `sortByLoc` function is exposed in @glimmer/syntax. An idea is to use it here in template-recast to replace a very similar function (see https://github.com/glimmerjs/glimmer-vm/pull/1266#discussion_r574910296).

This PR is a possible implementation. But IMO an other implementation should be done. Indeed, this PR would add another dependency and would cause a slow down.

See commit messages for more details.

## Links

- follow up of https://github.com/ember-template-lint/ember-template-recast/pull/627#discussion_r737815908
- implements https://github.com/glimmerjs/glimmer-vm/pull/1266#discussion_r574910296

### CODE REVIEW

File: src/parse-result.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { preprocess, builders, print as _print, traverse, ASTv1 as AST } from '@glimmer/syntax';
 import { getLines, sortByLoc, sourceForLoc } from './utils';
+import { klona } from 'klona/full';
```
Comment (rwjblue) [inline src/parse-result.ts:3]: How is this related to the changes to move to use `sortByLoc` from `@glimmer/syntax`?

Also, FWIW, the `sortByLoc` that landed in `@glimmer/syntax` was _originally_ extracted from us 😄

Comment (rwjblue) [reply src/parse-result.ts:3]: Also, wouldn't we want to use `klona/json`? Or do we need other structures now?

Comment (dcyriller) [reply src/parse-result.ts:3]: The sortByLoc function expects the `node.loc` property to be an instance of the SourceSpan class. But `JSON.parse(JSON.stringify())` and `klona/json` can only clone JSON data types. When cloned with JSON.parse(JSON.stringify()), the data structure is simplified (to a `SourceLocation`?). So sortByLoc fails on [`startPosition`](https://github.com/glimmerjs/glimmer-vm/blob/4f1bef0d9a8a3c3ebd934c5b6e09de4c5f6e4468/packages/%40glimmer/syntax/lib/generation/util.ts#L65) being undefined.

Comment (dcyriller) [reply src/parse-result.ts:3]: I am closing the PR as I don't have a good solution for now.
